hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/pipeline/Pipeline.java

```diff
@@ -287,11 +287,11 @@ public DatanodeDetails getClosestNode(Set<DatanodeDetails> excluded)
     if (excluded == null) {
       excluded = Collections.emptySet();
     }
-    if (nodesInOrder.get() == null || nodesInOrder.get().isEmpty()) {
+    if (nodesInOrder == null || nodesInOrder.isEmpty()) {
       LOG.debug("Nodes in order is empty, delegate to getFirstNode");
       return getFirstNode(excluded);
     }
-    for (DatanodeDetails d : nodesInOrder.get()) {
+    for (DatanodeDetails d : nodesInOrder) {
```

Comment:
When using `volatile` member repeatedly in the same method, I think it's recommended to assign it to a local variable:

 * avoid the cost of repeated `volatile` access
 * ensure the same instance is used in all statements

Reply:
will use list to update and access, updated